worlds/dark_souls_3/__init__.py

```diff
@@ -0,0 +1,281 @@
+        farron_keep_region.exits.append(Entrance(self.player, "Goto Carthus catacombs", farron_keep_region))
+        self.world.get_entrance("Goto Carthus catacombs", self.player).connect(catacombs_of_carthus_region)
+        catacombs_of_carthus_region.exits.append(Entrance(self.player, "Goto Irithyll of the boreal",
+                                                          catacombs_of_carthus_region))
+        catacombs_of_carthus_region.exits.append(Entrance(self.player, "Goto Smouldering Lake",
+                                                          catacombs_of_carthus_region))
+        self.world.get_entrance("Goto Irithyll of the boreal", self.player).\
+            connect(irithyll_of_the_boreal_valley_region)
+        self.world.get_entrance("Goto Smouldering Lake", self.player).connect(smouldering_lake_region)
+        irithyll_of_the_boreal_valley_region.exits.append(Entrance(self.player, "Goto Irithyll dungeon",
+                                                                   irithyll_of_the_boreal_valley_region))
+        irithyll_of_the_boreal_valley_region.exits.append(Entrance(self.player, "Goto Anor Londo",
+                                                                   irithyll_of_the_boreal_valley_region))
+        self.world.get_entrance("Goto Irithyll dungeon", self.player).connect(irithyll_dungeon_region)
+        self.world.get_entrance("Goto Anor Londo", self.player).connect(anor_londo_region)
+        irithyll_dungeon_region.exits.append(Entrance(self.player, "Goto Archdragon peak", irithyll_dungeon_region))
+        irithyll_dungeon_region.exits.append(Entrance(self.player, "Goto Profaned capital", irithyll_dungeon_region))
+        self.world.get_entrance("Goto Archdragon peak", self.player).connect(archdragon_peak_region)
+        self.world.get_entrance("Goto Profaned capital", self.player).connect(profaned_capital_region)
+        lothric_castle_region.exits.append(Entrance(self.player, "Goto Consumed King Garden", lothric_castle_region))
+        lothric_castle_region.exits.append(Entrance(self.player, "Goto Grand Archives", lothric_castle_region))
+        self.world.get_entrance("Goto Consumed King Garden", self.player).connect(consumed_king_garden_region)
+        self.world.get_entrance("Goto Grand Archives", self.player).connect(grand_archives_region)
+        consumed_king_garden_region.exits.append(Entrance(self.player, "Goto Untended Graves",
+                                                          consumed_king_garden_region))
+        self.world.get_entrance("Goto Untended Graves", self.player).connect(untended_graves_region)
+
+    # For each region, add the associated locations retrieved from the corresponding location_table
+    def create_region(self, name, location_table) -> Region:
+        new_region = Region(name, RegionType.Generic, name, self.player)
+        if location_table is not None:
+            for name, address in self.location_name_to_id.items():
```

Comment:
why not do this the other way around and just iterate through the supplied location_table instead of looping through the entirety of all locations for each region?